mono/utils/w32handle.c

```diff
@@ -1346,20 +1346,31 @@ mono_w32handle_wait_multiple (gpointer *handles, gsize nhandles, gboolean waital
 
 				if (i < lowest)
 					lowest = i;
+
+				if (!waitall)
```

Comment:
Why do we need that? That would lead to owning only the first signalled handle, and not any subsequent one which may also be signalled.